test/deflateparams.c

```diff
@@ -0,0 +1,254 @@
+#include <stdio.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <string.h>
+#include <assert.h>
+#include <stdlib.h>
+#include <inttypes.h>
+
+#include "zbuild.h"
+#ifdef ZLIB_COMPAT
+#  include "zlib.h"
+#else
+#  include "zlib-ng.h"
+#endif
+
+#if MAX_MEM_LEVEL >= 8
+#  define DEF_MEM_LEVEL 8
+#else
+#  define DEF_MEM_LEVEL  MAX_MEM_LEVEL
+#endif
+
+#define CHECK_ERR(err, msg) { \
+    if (err != Z_OK) { \
+        fprintf(stderr, "%s error: %d\n", msg, err); \
+        exit(1); \
+    } \
+}
+
+static alloc_func zalloc = NULL;
+static free_func zfree = NULL;
+
+/* ===========================================================================
+ * deflate() using small buffers
+ */
+void deflate_params_small(unsigned char *buf, int32_t len, int level, int window_bits, int mem_level, int strategy)
+{
+    PREFIX3(stream) c_stream; /* compression stream */
+    unsigned char compr[4096];
+    int err;
+
+    c_stream.zalloc = zalloc;
+    c_stream.zfree = zfree;
+    c_stream.opaque = (void *)0;
+    c_stream.total_in = 0;
+    c_stream.total_out = 0;
```

Comment:
This is something that would benefit from using a macro to eliminate/shorten very long span of repeated lines... maybe INIT_STREAM(...)... 

We will always use zalloc/zfree functions...

Reply:
I think we talked about doing that at some point in another thread. But I'm not so sure about how useful it would be.

Reply:
This is an issue that rises only with C code, as C++ has constructors that initialize the members with default values... I would rather not disable some warnings in external services just because the code has a lot of duplication.